pkg/template/controller/readiness.go

```diff
@@ -62,105 +75,109 @@ func checkBuildConfigReadiness(oc buildclient.Interface, obj runtime.Object) (bo
 // neither.
 func checkDeploymentReadiness(obj runtime.Object) (bool, bool, error) {
-	d := obj.(*extensions.Deployment)
+	d, ok := obj.(*kappsv1.Deployment)
+	if !ok {
+		return false, false, fmt.Errorf("object %T is not v1.Deployment", obj)
+	}
 
-	var progressing, available *extensions.DeploymentCondition
+	var progressing, available *kappsv1.DeploymentCondition
 	for i, condition := range d.Status.Conditions {
 		switch condition.Type {
-		case extensions.DeploymentProgressing:
+		case kappsv1.DeploymentProgressing:
 			progressing = &d.Status.Conditions[i]
 
-		case extensions.DeploymentAvailable:
+		case kappsv1.DeploymentAvailable:
 			available = &d.Status.Conditions[i]
 		}
 	}
 
 	ready := d.Status.ObservedGeneration == d.Generation &&
 		progressing != nil &&
-		progressing.Status == kapi.ConditionTrue &&
+		progressing.Status == corev1.ConditionTrue &&
 		progressing.Reason == deploymentutil.NewRSAvailableReason &&
 		available != nil &&
-		available.Status == kapi.ConditionTrue
+		available.Status == corev1.ConditionTrue
 
 	failed := d.Status.ObservedGeneration == d.Generation &&
 		progressing != nil &&
-		progressing.Status == kapi.ConditionFalse
+		progressing.Status == corev1.ConditionFalse
 
 	return ready, failed, nil
 }
 
 // checkDeploymentConfigReadiness determins if a DeploymentConfig is ready,
 // failed or neither.
 func checkDeploymentConfigReadiness(obj runtime.Object) (bool, bool, error) {
-	dc := obj.(*appsapi.DeploymentConfig)
+	dc, ok := obj.(*appsv1.DeploymentConfig)
+	if !ok {
+		return false, false, fmt.Errorf("object %T is not v1.DeploymentConfig", obj)
+	}
 
-	var progressing, available *appsapi.DeploymentCondition
+	var progressing, available *appsv1.DeploymentCondition
 	for i, condition := range dc.Status.Conditions {
 		switch condition.Type {
-		case appsapi.DeploymentProgressing:
+		case appsv1.DeploymentProgressing:
 			progressing = &dc.Status.Conditions[i]
 
-		case appsapi.DeploymentAvailable:
+		case appsv1.DeploymentAvailable:
 			available = &dc.Status.Conditions[i]
 		}
 	}
 
 	ready := dc.Status.ObservedGeneration == dc.Generation &&
 		progressing != nil &&
-		progressing.Status == kapi.ConditionTrue &&
-		progressing.Reason == appsapi.NewRcAvailableReason &&
+		progressing.Status == corev1.ConditionTrue &&
+		progressing.Reason == appsutil.NewRcAvailableReason &&
 		available != nil &&
-		available.Status == kapi.ConditionTrue
+		available.Status == corev1.ConditionTrue
 
 	failed := dc.Status.ObservedGeneration == dc.Generation &&
 		progressing != nil &&
-		progressing.Status == kapi.ConditionFalse
+		progressing.Status == corev1.ConditionFalse
 
 	return ready, failed, nil
 }
 
 // checkJobReadiness determins if a Job is ready, failed or neither.
 func checkJobReadiness(obj runtime.Object) (bool, bool, error) {
-	job := obj.(*batch.Job)
-
-	ready := job.Status.CompletionTime != nil
-	failed := job.Status.Failed > 0
-
-	return ready, failed, nil
+	job, ok := obj.(*batchv1.Job)
```

Comment:
you'll need all versions of jobs

Reply:
```
batch/v1/types.go
28:type Job struct {
```

It seems it only exists in batchv1... @soltysh can you confirm?

Reply:
Yup, that is correct.